decidim-elections/app/controllers/decidim/elections/admin/elections_controller.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Elections
+    module Admin
+      class ElectionsController < Admin::ApplicationController
+        include Decidim::Admin::HasTrashableResources
+        include Decidim::ApplicationHelper
+        include Decidim::Elections::Admin::Filterable
+
+        helper_method :elections
+
+        def index
+          enforce_permission_to :read, :election
+        end
+
+        def new
+          enforce_permission_to :create, :election
+          @form = form(Decidim::Elections::Admin::ElectionForm).from_params(
+            attachment: form(AttachmentForm).from_params({})
+          )
+        end
+
+        def create
+          enforce_permission_to :create, :election
+
+          @form = form(Decidim::Elections::Admin::ElectionForm).from_params(params, current_component:)
+
+          CreateElection.call(@form) do
+            on(:ok) do
+              flash[:notice] = I18n.t("elections.create.success", scope: "decidim.elections.admin")
+              redirect_to elections_path
+            end
+
+            on(:invalid) do
+              flash.now[:alert] = I18n.t("elections.create.invalid", scope: "decidim.elections.admin")
+              render action: "new"
+            end
+          end
+        end
+
+        def edit
+          enforce_permission_to(:update, :election, election:)
+          @form = form(Decidim::Elections::Admin::ElectionForm).from_model(election)
+        end
+
+        def update
```

Comment:
Just seen that if you have a validation error in page, you loose the uploaded files.